Remotion/Data/DomainObjects/Persistence/Rdbms/DbCommandBuilders/StoredProcedureDbCommandBuilder.cs

```diff
@@ -0,0 +1,66 @@
+// Copyright (c) rubicon IT GmbH, www.rubicon.eu
+// 
+// The re-motion Core Framework is free software; you can redistribute it 
+// and/or modify it under the terms of the GNU Lesser General Public License 
+// as published by the Free Software Foundation; either version 2.1 of the 
+// License, or (at your option) any later version.
+// 
+// re-motion is distributed in the hope that it will be useful, 
+// but WITHOUT ANY WARRANTY; without even the implied warranty of 
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the 
+// GNU Lesser General Public License for more details.
+// 
+// You should have received a copy of the GNU Lesser General Public License
+// along with re-motion; if not, see http://www.gnu.org/licenses.
+// 
+using System;
+using System.Collections.Generic;
+using System.Data;
+using Remotion.Data.DomainObjects.Queries;
+using Remotion.Data.DomainObjects.Queries.Configuration;
+using Remotion.Utilities;
+
+namespace Remotion.Data.DomainObjects.Persistence.Rdbms.DbCommandBuilders;
+
+/// <summary>
+/// Builds an <see cref="IDbCommand"/> for a given <see cref="QueryStatementType.StoredProcedure"/> call.
+/// </summary>
+public class StoredProcedureDbCommandBuilder : DbCommandBuilder
+{
+  public string ProcedureName { get; }
+  public IEnumerable<QueryParameterWithDataParameterDefinition> Parameters { get; }
+
+  public StoredProcedureDbCommandBuilder (string procedureName, IEnumerable<QueryParameterWithDataParameterDefinition> parameters, ISqlDialect sqlDialect)
+      : base(sqlDialect)
+  {
+    ArgumentUtility.CheckNotNullOrEmpty(nameof(procedureName), procedureName);
+    ArgumentUtility.CheckNotNull(nameof(parameters), parameters);
```

Comment:
Update you missed during vacation, sorry:

we have switched to ArgumentException.ThrowIfNullOrEmpty and ArgumentNullException.ThrowIfNull instead of those "standard" argument checks. please update the file. the other changes look okay.
There's no good way to guard against using the old syntax. I am still considering making them obsolete only in re-motion, though.